### PR TITLE
Added Results parsing for Heltour.

### PR DESCRIPTION
**Pull request by endrawes0**
refactored results related code from spreadsheet.js into results.js
updated chesster.js to use heltour apis
renamed heltour updateResult to updatePairing

### CODE REVIEW

File: src/chesster.js [hunk 1]
```diff
@@ -61,6 +62,19 @@ var chesster = new slack.Bot({
     config_file: config_file
 });
 
+function handleHeltourErrors(error){
+    if (_.isEqual(updatePairingResult['error'], "no_matching_rounds")) {
```
Comment (lakinwecker) [inline src/chesster.js]: this should refer to `error` and not `updatePairingResult['error']` right?

File: src/heltour.js [hunk 2]
```diff
@@ -1,16 +1,20 @@
 //------------------------------------------------------------------------------
 // Heltour related facilities
 //------------------------------------------------------------------------------
-
+const _ = require('lodash');
 const url = require('url');
 const http = require("./http.js");
 
-function findPairing(heltourConfig, white, black) {
+function findPairing(heltourConfig, white, black, league_tag) {
```
Comment (lakinwecker) [inline src/heltour.js:13]: Technically, `heltourConfig` has a league_tag in it. it comes from the config. I am unsure about whether we should use it or explicitly pass in the league tag.  I guess explicitly passing it in allows us to _not_ pass it in when we want to get pairings for both leagues.  

Let's document that here.

File: src/chesster.js (hunk 1 above)
Comment (endrawes0) [reply src/chesster.js]: yes

File: src/heltour.js (hunk 2 above)
Comment (endrawes0) [reply src/heltour.js:13]: ill add a comment. the intent was to allow calling findPairing without it - as you noted, this cannot be done by relying on the config.

File: src/chesster.js [hunk 3]
```diff
@@ -711,62 +731,63 @@ function(bot, message) {
...
             return;
         }
+        
+        heltour.findPairing(
+            heltourOptions,
+            result.white.name,
+            result.black.name,
+            heltourOptions.league_tag
+        ).then(function(findPairingResult){
+debugger;
```
Comment (lakinwecker) [inline src/chesster.js]: This can be removed.

File: src/chesster.js [hunk 4]
```diff
@@ -942,54 +963,44 @@ function processGameDetails(bot, message, details, options, heltourOptions){
...
-            } else {
-                replyGenericFailure(bot, message, "@endrawes0");
-                throw new Error("Error updating scheduling sheet: " + err);
-            }
+        if (updatePairingResult['error']) {
+            handleHeltourErrors(updatePairingResult['error']);
+            return; //if there was a problem with heltour, we should not take further steps 
         }
-        resultReplyUpdated(bot, message, updateResultsResult);
+        resultReplyUpdated(bot, message, updatePairingResult);
```
Comment (lakinwecker) [inline src/chesster.js:979]: A much better variable name. :P

File: src/chesster.js (hunk 3 above)
Comment (endrawes0) [reply src/chesster.js]: yup - thanks.

**Comment by lakinwecker** [comment]
Good work!

**Pull request merged**